cmake/external/icu.cmake

```diff
@@ -0,0 +1,178 @@
+include(FetchContent)
+
+set(ICU_TARGET_NAME "icu_external")
+set(ICU_URL https://github.com/unicode-org/icu/releases/download/release-70-1/icu4c-70_1-src.tgz)
+set(ICU_URL_HASH SHA256=8d205428c17bf13bb535300669ed28b338a157b1c01ae66d31d0d3e2d47c3fd5)
+
+set(THIRD_PARTY_PATH ${CMAKE_BINARY_DIR}/_deps/icu)
+set(ICU_SOURCE_DIR  ${THIRD_PARTY_PATH}/icu-src CACHE PATH "Path to extracted ICU source directory")
+set(ICU_BINARY_DIR  ${THIRD_PARTY_PATH}/icu-build CACHE PATH "Path to extracted ICU binary directory")
+set(ICU_INSTALL_DIR ${THIRD_PARTY_PATH}/icu-install CACHE PATH "Path to extracted ICU install directory")
+
+set(ICU_STATIC TRUE)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -std=c++11")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+
+if(OPENVINO_RUNTIME_COMPILE_DEFINITIONS)
+    foreach(def ${OPENVINO_RUNTIME_COMPILE_DEFINITIONS})
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D${def}")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D${def}")
+    endforeach()
+endif()
+
+set(HOST_ENV_CMAKE ${CMAKE_COMMAND} -E env
+        CC=${CMAKE_C_COMPILER}
+        CXX=${CMAKE_CXX_COMPILER}
```

Comment:
current issue on macOS is because of we override compilers (tested locally)

maybe we can skip overriding it temporarily? because it's not often situation when compiler on macOS is other than default one.